test/functional/services/data_grid.ts

```diff
@@ -275,6 +277,19 @@ export class DataGridService extends FtrService {
     return Promise.resolve(textArr);
   }
 
+  public async getControlColumnHeaderFields(): Promise<string[]> {
+    const result = await this.find.allByCssSelector(
+      '.euiDataGridHeaderCell--controlColumn > .euiDataGridHeaderCell__content'
+    );
+
+    const textArr = [];
+    for (const cell of result) {
+      const cellText = await cell.getVisibleText();
+      textArr.push(cellText.trim());
+    }
+    return Promise.resolve(textArr);
```

Comment:
**nit**: being the function `async`, it will automatically wrap the result in a new promise, you can remove the `Promise.resolve` usage.